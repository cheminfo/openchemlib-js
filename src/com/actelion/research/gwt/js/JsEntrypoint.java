package com.actelion.research.gwt.js;

import com.google.gwt.core.client.EntryPoint;

public class JsEntrypoint implements EntryPoint {
  @Override
  public void onModuleLoad() {}
}
