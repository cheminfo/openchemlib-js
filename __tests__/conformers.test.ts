import { assert, beforeAll, describe, expect, it } from 'vitest';

import { ConformerGenerator, Molecule, Resources } from '../lib/index.js';

beforeAll(() => {
  Resources.registerFromNodejs();
});

describe('ConformerGenerator', () => {
  it('should return one conformer', () => {
    const mol = Molecule.fromSmiles('COCCON');

    const gen = new ConformerGenerator(1);
    expect(gen.getConformerCount()).toBe(0);

    expect(mol.getAtomZ(0)).toBe(0);
    const result = gen.getOneConformerAsMolecule(mol);
    expect(gen.getConformerCount()).toBe(1);
    expect(result).toBe(mol);
    expect(mol.getAtomZ(0)).not.toBe(0);
  });

  it('should generate conformers', () => {
    const mol = Molecule.fromSmiles('COCCON');

    const gen = new ConformerGenerator(1);
    expect(gen.getPotentialConformerCount()).toBe(1);
    expect(gen.getConformerCount()).toBe(0);
    gen.initializeConformers(mol);
    expect(gen.getPotentialConformerCount()).toBeGreaterThan(1);

    const conf1 = gen.getNextConformerAsMolecule();
    expect(gen.getConformerCount()).toBe(1);
    assert(conf1);
    const conf1Molfile = conf1.toMolfile();
    expect(conf1Molfile).toMatchSnapshot();

    const conf2 = gen.getNextConformerAsMolecule();
    expect(gen.getConformerCount()).toBe(2);
    assert(conf2);
    const conf2Molfile = conf2.toMolfile();
    expect(conf2Molfile).toMatchSnapshot();

    expect(conf2).not.toBe(conf1);
    expect(conf2Molfile).not.toBe(conf1Molfile);
  });

  it('should use and return the passed molecule', () => {
    const mol = Molecule.fromSmiles('COCCON');

    const gen = new ConformerGenerator(1);
    gen.initializeConformers(mol);

    expect(mol.getAtomZ(0)).toBe(0);

    const conf = gen.getNextConformerAsMolecule(mol);
    expect(conf).toBe(mol);
    expect(mol.getAtomZ(0)).not.toBe(0);
  });

  it('should molecules() return an iterator', () => {
    const mol = Molecule.fromSmiles('COCCON');

    const gen = new ConformerGenerator(1);
    gen.initializeConformers(mol);

    const allConformers = [...gen.molecules()];
    expect(allConformers).toHaveLength(26);

    // All molecules should be different
    expect(new Set(allConformers).size).toBe(26);

    expect(allConformers[0].getAtomZ(0)).not.toBe(allConformers[1].getAtomZ(1));

    expect(allConformers.every((conf) => conf instanceof Molecule)).toBe(true);
  });
});
