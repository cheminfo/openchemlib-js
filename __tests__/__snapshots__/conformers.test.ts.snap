// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConformerGenerator > should generate conformers 1`] = `
"#1
Actelion Java MolfileCreator 1.0

 15 14  0  0  0  0  0  0  0  0999 V2000
    3.3997    1.5180   -1.6878 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.1300    1.8178   -0.4830 O   0  0  0  0  0  0  0  0  0  0  0  0
    3.2603    2.0511    0.6440 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.7851    3.5307    0.6291 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.9125    3.7446   -0.5003 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.6265    3.1403   -0.2603 N   0  0  0  0  0  0  0  0  0  0  0  0
    2.6991    0.7270   -1.5191 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.8757    2.3876   -2.0234 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.0865    1.2112   -2.4486 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.7860    1.8584    1.5546 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.4070    1.4077    0.5876 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.6383    4.1723    0.5573 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.2605    3.7450    1.5364 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.0145    3.4596   -0.8748 H   0  0  0  0  0  0  0  0  0  0  0  0
    0.6967    2.2226   -0.3494 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  1  7  1  0  0  0  0
  1  8  1  0  0  0  0
  1  9  1  0  0  0  0
  3 10  1  0  0  0  0
  3 11  1  0  0  0  0
  4 12  1  0  0  0  0
  4 13  1  0  0  0  0
  6 14  1  0  0  0  0
  6 15  1  0  0  0  0
M  END
"
`;

exports[`ConformerGenerator > should generate conformers 2`] = `
"#2
Actelion Java MolfileCreator 1.0

 15 14  0  0  0  0  0  0  0  0999 V2000
    4.6113    0.4611   -0.5340 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.1300    1.8178   -0.4830 O   0  0  0  0  0  0  0  0  0  0  0  0
    3.2603    2.0511    0.6440 C   0  0  0  0  0  0  0  0  0  0  0  0
    2.7851    3.5307    0.6291 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.9051    4.3988    0.9019 O   0  0  0  0  0  0  0  0  0  0  0  0
    3.4838    5.7769    0.8923 N   0  0  0  0  0  0  0  0  0  0  0  0
    5.0424    0.1767    0.4032 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.8070   -0.2035   -0.7682 H   0  0  0  0  0  0  0  0  0  0  0  0
    5.3601    0.3816   -1.2940 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.7860    1.8584    1.5546 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.4070    1.4077    0.5876 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.0301    3.6641    1.3754 H   0  0  0  0  0  0  0  0  0  0  0  0
    2.3740    3.7625   -0.3309 H   0  0  0  0  0  0  0  0  0  0  0  0
    4.1394    6.3041    1.2749 H   0  0  0  0  0  0  0  0  0  0  0  0
    3.3448    6.0497    0.0198 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  1  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  5  6  1  0  0  0  0
  1  7  1  0  0  0  0
  1  8  1  0  0  0  0
  1  9  1  0  0  0  0
  3 10  1  0  0  0  0
  3 11  1  0  0  0  0
  4 12  1  0  0  0  0
  4 13  1  0  0  0  0
  6 14  1  0  0  0  0
  6 15  1  0  0  0  0
M  END
"
`;
